src/axom/quest/MeshViewUtil.hpp

```diff
@@ -0,0 +1,778 @@
+      for(int d = 0; d < DIM; ++d)
+      {
+        counts[d] += onVertex;
+      }
+      auto rval1 = rval;
+      rval = rval1.subspan(offsets, counts);
+    }
+
+    return rval;
+  }
+  //@}
+
+  //@{
+  //!@name Creating data
+
+  /*!
+    @brief Create a new scalar nodal data field.
+    @param [in] fieldName
+    @param [in] association "vertex" or "element"
+    @param [in] dtype Conduit data type to put in the field.  Must be at least
+                big enough for the strides specified.
+    @param [in] strides Data strides.  Set to zero for no ghosts and default strides.
+    @param [in] offsets Data index offsets.
+
+    Field data allocation is done by Conduit, so the data lives in
+    host memory.  Conduit currently doesn't provide a means to allocate
+    the array in device memory.
+  */
+  void createField(const std::string& fieldName,
+                   const std::string& association,
+                   const conduit::DataType& dtype,
+                   const axom::StackArray<axom::IndexType, DIM>& strides,
+                   const axom::StackArray<axom::IndexType, DIM>& offsets)
+  {
+    if(m_dom->has_path("fields/" + fieldName))
+    {
+      SLIC_ERROR(
+        axom::fmt::format("Cannot create field {}.  It already exists.",
+                          fieldName));
+    }
+    if(association != "vertex" && association != "element")
+    {
+      SLIC_ERROR(
+        axom::fmt::format("Not yet supporting association '{}'.", association));
+    }
+
+    auto realExtents = getRealExtents(association);
+
+    conduit::Node& fieldNode = m_dom->fetch("fields/" + fieldName);
+    fieldNode["association"] = association;
+    fieldNode["topology"] = m_topologyName;
+
+    {
+      fieldNode["strides"].set(conduit::DataType::int32(DIM));
+      std::int32_t* tmpPtr = fieldNode["strides"].as_int32_ptr();
+      for(int d = 0; d < DIM; ++d)
+      {
+        tmpPtr[d] = strides[d];
+      }
+    }
+
+    {
+      fieldNode["offsets"].set(conduit::DataType::int32(DIM));
+      std::int32_t* tmpPtr = fieldNode["offsets"].as_int32_ptr();
+      for(int d = 0; d < DIM; ++d)
+      {
+        tmpPtr[d] = offsets[d];
+      }
+    }
+
+    axom::IndexType slowDir = 0;
+    for(int d = 0; d < DIM; ++d)
+    {
+      if(strides[slowDir] < strides[d])
+      {
+        slowDir = d;
+      }
+    }
+    auto extras = dtype.number_of_elements() -
+      strides[slowDir] * (offsets[slowDir] + realExtents[slowDir]);
+    if(extras < 0)
+    {
+      SLIC_ERROR(
+        axom::fmt::format("Insufficient space allocated for data, given the "
+                          "offsets and strides."));
+    }
+
+    fieldNode["values"].set(dtype);
+  }
+  //@}
+
+private:
+  conduit::Node* m_dom = nullptr;
+  conduit::Node* m_topology = nullptr;
+  conduit::Node* m_coordset = nullptr;
+  const conduit::Node* m_cdom = nullptr;
+  const conduit::Node* m_ctopology = nullptr;
+  const conduit::Node* m_ccoordset = nullptr;
```

Comment:
Would these be better as paths within the parent conduit node rather than pointers to conduit nodes?

Reply:
Either would work.  I chose this way because the paths made for longer lines that appear repetitively many places.